app/src/main/java/com/codesoom/assignment/application/ProductCommandService.java

```diff
@@ -0,0 +1,85 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.controller.dto.ProductUpdateRequest;
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.exception.ProductNotFoundException;
+import com.codesoom.assignment.repository.ProductRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * 상품 생성/수정/삭제를 처리합니다.
+ */
+@Service
+@Transactional
+public class ProductCommandService {
+
+    private final ProductRepository productRepository;
+
+    public ProductCommandService(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    /**
+     * 상품을 생성하여 반환합니다.
+     * @param source 생성할 상품
+     * @return 생성된 상품
+     */
+    public Product create(ProductUpdateRequest source) {
+        Product product = new Product(source.getName(),
+                source.getMaker(),
+                source.getPrice(),
+                source.getImageUrl()
+        );
+
+        return productRepository.save(product);
+    }
+
+    /**
+     * 상품을 수정하여 반환합니다.
+     * @Param id 수정할 상품 아이디
+     * @param source 수정할 상품
+     * @return 수정된 상품
+     * @throws ProductNotFoundException 수정할 상품을 찾지 못한 경우
+     */
+    public Product update(Long id, ProductUpdateRequest source) {
+        Product product = productRepository.findById(id)
+                .orElseThrow(() -> new ProductNotFoundException(id + "에 해당하는 상품을 찾지 못해 업데이트할 수 없습니다."));
+
+        product.update(source.getName(),
+                source.getMaker(),
+                source.getPrice(),
+                source.getImageUrl()
+        );
+
+        return product;
+    }
+
+    /**
+     * 상품 하나를 삭제합니다.
+     * @param id 삭제할 상품 아이디
+     * @throws ProductNotFoundException 삭제할 상품을 찾지 못한 경우
+     */
+    public void deleteById(Long id) {
+        Product product = productRepository.findById(id)
+                .orElseThrow(() -> new ProductNotFoundException(id + "에 해당하는 상품을 찾지 못해 삭제할 수 없습니다."));
+
+        productRepository.delete(product);
+    }
+
+    /**
+     * 상품 여러 개를 삭제합니다.
+     * @param ids 삭제할 상품 아이디 목록
+     * @throws ProductNotFoundException 삭제할 상품을 찾지 못한 경우
+     */
+    public void deleteByIds(List<Long> ids) {
```

Comment:
이렇게 하는 것도 고려해 볼 수 있겠죠.
```suggestion
    public void deleteByIds(Set<Long> ids) {
```

Reply:
중복을 방지할 수 있어서 좋은 것 같아요!

Reply:
중복 아이디로 인한 예상치 못한 예외 던짐을 방지할 수도 있을 거에요. `for` loop를 돌려서 삭제를 하고 있었다면 중복 아이디 리스트가 주어졌을 때 문제가 생겼겠죠.